src/cstar-cli/cstar/cli/bpd/awardwinner.py

```diff
@@ -182,18 +182,62 @@ def send_jackpot_winners(self):
   def read_state_41(self):
     transfers_df = pd.read_csv(self.args.file, sep=';')
     transfers_df = transfers_df.apply(self._extract_transfer_id, axis=1)
-    print(transfers_df.head(5).to_csv(index=False))
+    assert transfers_df.idKey.size == transfers_df.idKey.unique().size, "idKey not unique"
+
+    print(transfers_df)
+
+    pagopa_transfers_q = self.db_connection.cursor().mogrify(
+      "SELECT id_n, insert_date_t, insert_user_s, update_date_t, "
+      "update_user_s, enabled_b, technical_account_holder_s, " 
+      "chunk_filename_s, status_s, esito_bonifico_s, cro_s, "
+      "data_esecuzione_t, result_reason_s, to_notify_b, notify_times_n, "
+      "notify_id_s, ticket_id_n, related_id_n, consap_id_n, issuer_card_id_s "
+      "FROM bpd_citizen.bpd_award_winner " 
+      "WHERE id_n in %(id_n_list)s;",
+      {
+        "id_n_list": tuple(transfers_df.idKey.unique()),
+      }
+    )
+
+    pagopa_transfers_df = pd.read_sql(pagopa_transfers_q, self.db_connection)
+    pagopa_transfers_df['id_n'] = pagopa_transfers_df['id_n'].astype(str)
+
+    pagopa_n_tranfers_per_awp_q = self.db_connection.cursor().mogrify(
+      "SELECT baw.id_n, C.pagopa_n_tranfers_per_awp "
+      "FROM bpd_citizen.bpd_award_winner baw "
+      "INNER JOIN "
+      "(SELECT fiscal_code_s, award_period_id_n, count(fiscal_code_s) AS pagopa_n_tranfers_per_awp "
```

Comment:
If I understand correctly the query this line could be simplified as:

`"(SELECT count(fiscal_code_s) AS pagopa_n_tranfers_per_awp "`

Reply:
Maybe I'm wrong, but I think all the columns returned by that select are needed, because the fiscal code and the award period are used inside the conditions of the inner join in the external query. I'm trying to add - for the selected `id_n`s - a column which counts the number of wire transfers we tried for a given fiscal code in a given award period